appinventor/components-ios/src/ImageSprite.swift

```diff
@@ -126,8 +126,8 @@ open class ImageSprite: Sprite {
   
   override func updateDisplayLayer() {
     let d = DisplayLayer
-    let xCenter = CGFloat(XCenter)
-    let yCenter = CGFloat(YCenter)
+    let xCenter = CGFloat(XCenter) + Double(Width / 2)
+    let yCenter = CGFloat(YCenter) + Double(Height / 2)
```

Comment:
Rather than use XCenter and YCenter, use X and Y, otherwise if we fix the issue that causes XCenter and YCenter to be miscalculated later on this will break.